DirectProgramming/DPC++FPGA/Tutorials/Features/experimental/ip_authoring_mmhost/src/mmhost.cpp

```diff
@@ -0,0 +1,133 @@
+#include <CL/sycl.hpp>
+#include <iomanip>
+#include <sycl/ext/intel/experimental/interfaces.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+
+#include "dpc_common.hpp"
+
+using namespace cl::sycl;
+using ext::intel::experimental::property::usm::buffer_location;
+
+template <typename T>
+inline std::shared_ptr<T> make_malloc_shared(queue &q, int n, int kBL) {
+  T *mem =
+      malloc_shared<T>(sizeof(T) * n, q, property_list{buffer_location(kBL)});
+  return std::move(
+      std::shared_ptr<T>(mem, [&q](T *ptr) { sycl::free(ptr, q); }));
+}
+
+constexpr int BL1 = 1;
```

Comment:
Please use expanded names: kBufferLocation1 = 1; and similar for the constexpr expressions below